src/cennznut/mod.rs

```diff
@@ -87,15 +101,35 @@ impl Decode for CENNZnut {
     }
 }
 
-impl Validate<ModuleDomain> for CENNZnut {
+impl Validate<RuntimeDomain, ContractDomain> for CENNZnut {
     fn validate(
         &self,
         module_name: &str,
         method_name: &str,
         args: &[PactType],
-    ) -> Result<(), ValidationErr<ModuleDomain>> {
+    ) -> Result<(), ValidationErr<RuntimeDomain>> {
+        self.validate_runtime_call(module_name, method_name, args)
+    }
+
+    fn validate_runtime_call(
+        &self,
+        module_name: &str,
+        method_name: &str,
+        args: &[PactType],
+    ) -> Result<(), ValidationErr<RuntimeDomain>> {
+        match &self {
```

Comment:
validate_module, validate_contract and so on should belong to CENNZnut. We shouldn't check for the version everywhere.

Reply:
Unsure what is meant by this?

It is possible that we could move `validate` to be part of the versioned CennzNut, but this PR doesn't address that design decision